src/core/ext/client_config/lb_policy_factory.h

```diff
@@ -47,8 +47,25 @@ struct grpc_lb_policy_factory {
   const grpc_lb_policy_factory_vtable *vtable;
 };
 
+/** A resolved address alongside any LB related information associated with it.
+ * \a user_data, if not \a NULL, is opaque and meant to be consumed by the gRPC
+ * LB policy. Anywhere else, refer to the functions in \a
+ * grpc_lb_policy_user_data_vtable to operate with it */
+typedef struct grpc_lb_address {
+  grpc_resolved_address *resolved_address;
+  void *user_data;
+} grpc_lb_address;
+
+/** Functions acting upon the opaque \a grpc_lb_address.user_data */
+typedef struct grpc_lb_policy_user_data_vtable {
```

Comment:
The vtable approach was introduced when user_data wasn't an mdelem but something with a more complex structure. Long story short, your suggestion is perfect for the time being. If we ever need something more complex/flexible -like the vtable-, we can always make (re)introduce it later. Done.